lightning/src/ln/channel.rs

```diff
@@ -1656,6 +1657,54 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		cmp::min(self.value_to_self_msat as i64 - self.get_outbound_pending_htlc_stats().1 as i64, 0) as u64)
 	}
 
+	fn commit_tx_fee_outbound_htlc(&self) -> u64 {
+		if !self.channel_outbound {
+			return 0;
+		} else {
+			let mut their_acked_htlcs = self.pending_inbound_htlcs.len() as u64;
+
+			for ref htlc in self.pending_outbound_htlcs.iter() {
+				match htlc.state {
+					OutboundHTLCState::Committed => their_acked_htlcs +=1,
+					OutboundHTLCState::RemoteRemoved {..} => their_acked_htlcs +=1,
+					OutboundHTLCState::LocalAnnounced {..} => their_acked_htlcs += 1,
+					_ => {},
+				}
+			}
+
+			for ref htlc in self.holding_cell_htlc_updates.iter() {
+				match htlc {
+					&&HTLCUpdateAwaitingACK::AddHTLC { .. } => their_acked_htlcs += 1,
+					_ => {},
+				}
+			}
+
+			return self.feerate_per_kw * (COMMITMENT_TX_BASE_WEIGHT + their_acked_htlcs * COMMITMENT_TX_WEIGHT_PER_HTLC);
+		}
+	}
+
+	fn commit_tx_fee_inbound_htlc(&self) -> u64 {
+		if self.channel_outbound {
+			return 0;
+		} else {
+			let mut their_acked_htlcs = self.pending_inbound_htlcs.len() as u64;
+
+			// When calculating the set of HTLCs which will be included in their next
+			// commitment_signed, all inbound HTLCs are included (as all states imply it will be
+			// included) and only committed outbound HTLCs, see below.
+			for ref htlc in self.pending_outbound_htlcs.iter() {
+				if let OutboundHTLCState::Committed = htlc.state {
+					// We only include outbound HTLCs if it will not be included in their next
+					// commitment_signed, ie if they've responded to us with an RAA after announcement
+					// and if they haven't yet started removal.
+					their_acked_htlcs += 1;
+				}
+			}
+
+			return self.feerate_per_kw * (COMMITMENT_TX_BASE_WEIGHT + their_acked_htlcs * COMMITMENT_TX_WEIGHT_PER_HTLC);
```

Comment:
I think you need a +1 to include the next HTLC that we're adding/checking. Also, I think you need a /1000 - the per_kw stands for per_kilo_weight, and you're multiplying it by raw weight.

Reply:
Oh didn't see it at first parse ! Honestly, given how much we should be careful with weight/msat/sat, we should typify all of them, and let Rust compiler yells at us when we try to add apple and oranges ?

Reply:
Agreed w/ the +1 part.

Re: the /1000 -- IIUC that'll give me the fee in satoshis? so in theory I can leave it out if i want millisats?

Reply:
> Oh didn't see it at first parse ! Honestly, given how much we should be careful with weight/msat/sat, we should typify all of them, and let Rust compiler yells at us when we try to add apple and oranges ?

That's a good idea -- maybe follow-up PR? probably a pretty big changeset..

Reply:
> Re: the /1000 -- IIUC that'll give me the fee in satoshis? so in theory I can leave it out if i want millisats?

Hmm, true, but I think it needs to be rounded down, so would still need to do it.